Kickstarter-iOS/Features/ConfirmDetails/Views/ConfirmDetailsContinueCTAView.swift

```diff
@@ -0,0 +1,163 @@
+  private lazy var titleLabel: UILabel = { UILabel(frame: .zero) }()
+  private lazy var amountLabel: UILabel = { UILabel(frame: .zero) }()
+
+  private(set) lazy var continueButton: UIButton = {
+    UIButton(type: .custom)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  public weak var delegate: ConfirmDetailsContinueCTAViewDelegate?
+
+  // MARK: - Lifecycle
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+
+    self.configureSubviews()
+    self.setupConstraints()
+
+    self.continueButton.addTarget(
+      self, action: #selector(self.continueButtonTapped),
+      for: .touchUpInside
+    )
+  }
+
+  required init?(coder _: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: - Styles
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    _ = self
+      |> \.layoutMargins .~ .init(all: Styles.grid(3))
+
+    _ = self.layer
+      |> checkoutLayerCardRoundedStyle
+      |> \.backgroundColor .~ UIColor.ksr_white.cgColor
+      |> \.shadowColor .~ UIColor.ksr_black.cgColor
+      |> \.shadowOpacity .~ 0.12
+      |> \.shadowOffset .~ CGSize(width: 0, height: -1.0)
+      |> \.shadowRadius .~ CGFloat(1.0)
+      |> \.maskedCorners .~ [
+        CACornerMask.layerMaxXMinYCorner,
+        CACornerMask.layerMinXMinYCorner
+      ]
+
+    _ = self.rootStackView
+      |> verticalStackViewStyle
+      |> \.spacing .~ Styles.grid(3)
+
+    _ = self.titleAndAmountStackView
+      |> self.titleAndAmountStackViewStyle
+
+    _ = self.titleLabel
+      |> self.titleLabelStyle
+
+    _ = self.amountLabel
+      |> self.amountLabelStyle
+
+    _ = self.continueButton
+      |> greenButtonStyle
+      |> UIButton.lens.title(for: .normal) %~ { _ in Strings.Continue() }
+  }
+
+  // MARK: Functions
+
+  private func configureSubviews() {
+    _ = (self.rootStackView, self)
+      |> ksr_addSubviewToParent()
+      |> ksr_constrainViewToMarginsInParent()
+
+    _ = ([self.titleAndAmountStackView, self.continueButton], self.rootStackView)
+      |> ksr_addArrangedSubviewsToStackView()
+
+    _ = ([self.titleLabel, self.amountLabel], self.titleAndAmountStackView)
+      |> ksr_addArrangedSubviewsToStackView()
+  }
+
+  private func setupConstraints() {
+    NSLayoutConstraint.activate([
+      self.continueButton.heightAnchor.constraint(greaterThanOrEqualToConstant: Layout.Button.minHeight)
+    ])
+  }
+
+  func configure(with data: ConfirmDetailsContinueCTAViewData) {
+    if let attributedAmount = attributedCurrency(with: data.project, total: data.total) {
+      self.amountLabel.attributedText = attributedAmount
+    }
+  }
+
+  private func attributedCurrency(with project: Project, total: Double) -> NSAttributedString? {
+    let defaultAttributes = checkoutCurrencyDefaultAttributes()
+      .withAllValuesFrom([.foregroundColor: UIColor.ksr_support_700])
+    let projectCurrencyCountry = projectCountry(forCurrency: project.stats.currency) ?? project.country
```

Comment:
Do we expect this fall-through to ever have to happen? I wonder if we should just change the type of `project.currency` to be a forced non-nil value, since it would be pretty odd if we didn't have that information from the server.